src/test/unit/resources/test_detectors.py

```diff
@@ -127,3 +129,35 @@ def test_get_resource_aws_ecs_resource_detector_not_ecs_container(caplog):
     assert ResourceAttributes.CLOUD_PLATFORM not in resource.attributes
     assert ResourceAttributes.CONTAINER_NAME not in resource.attributes
     assert ResourceAttributes.CONTAINER_ID not in resource.attributes
+
+
+@patch(
+    "opentelemetry.sdk.extension.aws.resource.eks._get_k8s_cred_value",
+    return_value="MOCK_TOKEN",
+)
+@patch(
+    "opentelemetry.sdk.extension.aws.resource.eks._is_eks",
+    return_value=True,
+)
+@patch(
+    "opentelemetry.sdk.extension.aws.resource.eks._get_cluster_info",
+    return_value=get_cluster_info,
+)
+@patch(
+    "builtins.open",
+    new_callable=mock_open,
+    read_data=container_id_text,
+)
+def test_get_resource_aws_eks_resource_detector(
+    mock_open_function,
+    mock_get_cluster_info,
+    mock_is_eks,
+    mock_get_k8_cred_value,
+):
+    resource = get_resource({})
+
+    assert resource.attributes[ResourceAttributes.CLOUD_PROVIDER] == "aws"
+    assert resource.attributes[ResourceAttributes.CLOUD_PLATFORM] == "aws_eks"
+    assert isinstance(resource.attributes[ResourceAttributes.K8S_CLUSTER_NAME], str)
+    assert len(resource.attributes[ResourceAttributes.K8S_CLUSTER_NAME]) > 1
+    assert isinstance(resource.attributes[ResourceAttributes.CONTAINER_ID], str)
```

Comment:
Why not add
`assert len(resource.attributes[ResourceAttributes. CONTAINER_ID]) > 1`
as well?